course7/homeworks/patrickh/homework3/complex.cpp

```diff
@@ -0,0 +1,55 @@
+#include <iostream>
+#include "complex.h"
+
+Complex::Complex(double real, double imaginary) : mReal(real), mImaginary(imaginary)
+{
+
+}
+
+Complex::Complex(const std::string& input)
+{
+    size_t index = 0;
+    mReal = std::stod(input, &index);
+    if(input.substr(index, 2 ) == "+i")
+        mImaginary = std::stod(input.substr(index + 2));
+}
+
+Complex Complex::operator+(const Complex& other)
+{
+    return Complex(this->mReal + other.mReal, this->mImaginary + other.mImaginary);
+}
+
+Complex Complex::operator-(const Complex& other)
+{
+    return Complex(this->mReal - other.mReal, this->mImaginary - other.mImaginary);
+}
+
+Complex Complex::operator*(const Complex& other)
```

Comment:
It would be better if these small functions were implemented in the header file. Easier to read by humans, and easier to optimize by the compiler.